From 66841b7b640d53ea09a92221080f18e3397de21f Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Thu, 6 Oct 2022 12:04:54 +0300
Subject: [PATCH 05/49] s32: ddr: Switch to quote include statements

Changed the include statements to be quote-based, in order
to align to the DDR Tool generated format. This means that
the DDR driver folder has to be added to the PLAT_INCLUDES
directly.

Also if external DDR firmware is used, the CUSTOM_DDR_DRV
needs to be added to PLAT_INCLUDES too, before the normal
DDR Driver folder, since the -I<include> folders are searched
for headers in the order they were included, from left to right.

Issue: ALB-9354
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32/ddr/ddr_density.c             | 2 +-
 drivers/nxp/s32/ddr/ddr_init.c                | 3 +--
 drivers/nxp/s32/ddr/ddr_lp.c                  | 5 ++---
 drivers/nxp/s32/ddr/ddr_lp_csr.c              | 4 ++--
 drivers/nxp/s32/ddr/ddr_utils.c               | 3 +--
 drivers/nxp/s32/ddr/ddrss_cfg.c               | 3 +--
 drivers/nxp/s32/ddr/imem_cfg.c                | 4 ++--
 drivers/nxp/s32/ddr/s32g2/ddrc_cfg.c          | 2 +-
 drivers/nxp/s32/ddr/s32g2/dmem_cfg.c          | 4 ++--
 drivers/nxp/s32/ddr/s32g2/dq_swap_cfg.c       | 4 ++--
 drivers/nxp/s32/ddr/s32g2/phy_cfg.c           | 4 ++--
 drivers/nxp/s32/ddr/s32g2/pie_cfg.c           | 4 ++--
 drivers/nxp/s32/ddr/s32g3/ddrc_cfg.c          | 2 +-
 drivers/nxp/s32/ddr/s32g3/dmem_cfg.c          | 4 ++--
 drivers/nxp/s32/ddr/s32g3/dq_swap_cfg.c       | 4 ++--
 drivers/nxp/s32/ddr/s32g3/phy_cfg.c           | 4 ++--
 drivers/nxp/s32/ddr/s32g3/pie_cfg.c           | 4 ++--
 drivers/nxp/s32/ddr/s32r/ddrc_cfg.c           | 2 +-
 drivers/nxp/s32/ddr/s32r/dmem_cfg.c           | 2 +-
 drivers/nxp/s32/ddr/s32r/dq_swap_cfg.c        | 2 +-
 drivers/nxp/s32/ddr/s32r/phy_cfg.c            | 2 +-
 drivers/nxp/s32/ddr/s32r/pie_cfg.c            | 2 +-
 plat/nxp/s32/s32_bl2_el3.c                    | 2 +-
 plat/nxp/s32/s32_common.mk                    | 1 +
 plat/nxp/s32/s32_ddr.mk                       | 6 ++++++
 plat/nxp/s32/s32g/bl31_sram/bl31_sram.mk      | 2 +-
 plat/nxp/s32/s32g/bl31_ssram/bl31ssram_main.c | 2 +-
 plat/nxp/s32/s32g/s32g_bl2_el3.c              | 6 +++---
 plat/nxp/s32/s32r/s32r_bl2_el3.c              | 2 +-
 29 files changed, 47 insertions(+), 44 deletions(-)

diff --git a/drivers/nxp/s32/ddr/ddr_density.c b/drivers/nxp/s32/ddr/ddr_density.c
index 6378326bc..db495f879 100644
--- a/drivers/nxp/s32/ddr/ddr_density.c
+++ b/drivers/nxp/s32/ddr/ddr_density.c
@@ -32,7 +32,7 @@
 #include <lib/mmio.h>
 #include <lib/utils_def.h>
 #include <ddr/ddr_density.h>
-#include <ddr/ddr_utils.h>
+#include "ddr_utils.h"
 
 #define OFFSET_DDRC_START_ADDRMAP	0x21C
 
diff --git a/drivers/nxp/s32/ddr/ddr_init.c b/drivers/nxp/s32/ddr/ddr_init.c
index bc47759bc..b70b11e23 100644
--- a/drivers/nxp/s32/ddr/ddr_init.c
+++ b/drivers/nxp/s32/ddr/ddr_init.c
@@ -28,8 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
-#include <lib/mmio.h>
+#include "ddr_init.h"
 
 static uint32_t ddrc_init_cfg(const struct ddrss_config *config);
 static uint32_t execute_training(const struct ddrss_config *config);
diff --git a/drivers/nxp/s32/ddr/ddr_lp.c b/drivers/nxp/s32/ddr/ddr_lp.c
index ab236c1ee..aac778f8a 100644
--- a/drivers/nxp/s32/ddr/ddr_lp.c
+++ b/drivers/nxp/s32/ddr/ddr_lp.c
@@ -28,9 +28,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_lp.h>
-#include <ddr/ddr_init.h>
-#include <lib/mmio.h>
+#include "ddr_lp.h"
+#include "ddr_init.h"
 
 static void load_csr(uintptr_t load_from);
 static void load_ddrc_regs(uintptr_t load_from);
diff --git a/drivers/nxp/s32/ddr/ddr_lp_csr.c b/drivers/nxp/s32/ddr/ddr_lp_csr.c
index a49a17c0e..4d7e0235d 100644
--- a/drivers/nxp/s32/ddr/ddr_lp_csr.c
+++ b/drivers/nxp/s32/ddr/ddr_lp_csr.c
@@ -28,8 +28,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_lp.h>
-#include <ddr/ddr_init.h>
+#include "ddr_lp.h"
+#include "ddr_init.h"
 
 /* DDR PHY Configuration and Status registers */
 const uint32_t csr_to_store[] = {
diff --git a/drivers/nxp/s32/ddr/ddr_utils.c b/drivers/nxp/s32/ddr/ddr_utils.c
index b3a6ddfba..d31105a93 100644
--- a/drivers/nxp/s32/ddr/ddr_utils.c
+++ b/drivers/nxp/s32/ddr/ddr_utils.c
@@ -28,8 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_utils.h>
-#include <lib/mmio.h>
+#include "ddr_utils.h"
 
 #ifdef STORE_CSR_ENABLE
 /* Store Configuration Status Registers. */
diff --git a/drivers/nxp/s32/ddr/ddrss_cfg.c b/drivers/nxp/s32/ddr/ddrss_cfg.c
index 9e2531139..38f36a303 100644
--- a/drivers/nxp/s32/ddr/ddrss_cfg.c
+++ b/drivers/nxp/s32/ddr/ddrss_cfg.c
@@ -28,8 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <libc/string.h>
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct ddrss_config configs[] = {
 	{
diff --git a/drivers/nxp/s32/ddr/imem_cfg.c b/drivers/nxp/s32/ddr/imem_cfg.c
index 2984ea3b3..b7cbdc924 100644
--- a/drivers/nxp/s32/ddr/imem_cfg.c
+++ b/drivers/nxp/s32/ddr/imem_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 uint16_t imem_1d_cfg[] = {
 	0x0114,
diff --git a/drivers/nxp/s32/ddr/s32g2/ddrc_cfg.c b/drivers/nxp/s32/ddr/s32g2/ddrc_cfg.c
index 87eedbc31..e47dba283 100644
--- a/drivers/nxp/s32/ddr/s32g2/ddrc_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g2/ddrc_cfg.c
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf ddrc_cfg[] = {
 	{0x4007c604, 0x00000000U},
diff --git a/drivers/nxp/s32/ddr/s32g2/dmem_cfg.c b/drivers/nxp/s32/ddr/s32g2/dmem_cfg.c
index 0cabdb11d..49b4fbb00 100644
--- a/drivers/nxp/s32/ddr/s32g2/dmem_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g2/dmem_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2021 NXP
+ * Copyright 2020-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 uint16_t dmem_1d_cfg[] = {
 	0x0000,
diff --git a/drivers/nxp/s32/ddr/s32g2/dq_swap_cfg.c b/drivers/nxp/s32/ddr/s32g2/dq_swap_cfg.c
index 36be2cad8..90ea6678c 100644
--- a/drivers/nxp/s32/ddr/s32g2/dq_swap_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g2/dq_swap_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct dqconf dq_swap_cfg[] = {
 	{0x40394830, 0x00},
diff --git a/drivers/nxp/s32/ddr/s32g2/phy_cfg.c b/drivers/nxp/s32/ddr/s32g2/phy_cfg.c
index 2bcd6d628..7aeece92c 100644
--- a/drivers/nxp/s32/ddr/s32g2/phy_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g2/phy_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf_16 phy_cfg[] = {
 	{0x4038040c, 0x00000001},
diff --git a/drivers/nxp/s32/ddr/s32g2/pie_cfg.c b/drivers/nxp/s32/ddr/s32g2/pie_cfg.c
index 221e5adf2..fc23e6e27 100644
--- a/drivers/nxp/s32/ddr/s32g2/pie_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g2/pie_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf_16 pie_cfg[] = {
 	{0x40380c04, 0x00000010},
diff --git a/drivers/nxp/s32/ddr/s32g3/ddrc_cfg.c b/drivers/nxp/s32/ddr/s32g3/ddrc_cfg.c
index 164894c18..d808fdf9a 100644
--- a/drivers/nxp/s32/ddr/s32g3/ddrc_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g3/ddrc_cfg.c
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf ddrc_cfg[] = {
 	{0x403c0000, 0x83080020U},
diff --git a/drivers/nxp/s32/ddr/s32g3/dmem_cfg.c b/drivers/nxp/s32/ddr/s32g3/dmem_cfg.c
index a6ac709e4..727b6a41b 100644
--- a/drivers/nxp/s32/ddr/s32g3/dmem_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g3/dmem_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 uint16_t dmem_1d_cfg[] = {
 	0x0000,
diff --git a/drivers/nxp/s32/ddr/s32g3/dq_swap_cfg.c b/drivers/nxp/s32/ddr/s32g3/dq_swap_cfg.c
index 0972b9a8f..b635da444 100644
--- a/drivers/nxp/s32/ddr/s32g3/dq_swap_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g3/dq_swap_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct dqconf dq_swap_cfg[] = {
 	{0x40394830, 0x00},
diff --git a/drivers/nxp/s32/ddr/s32g3/phy_cfg.c b/drivers/nxp/s32/ddr/s32g3/phy_cfg.c
index 0f12daa19..562539372 100644
--- a/drivers/nxp/s32/ddr/s32g3/phy_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g3/phy_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf_16 phy_cfg[] = {
 	{0x4038040c, 0x00000001},
diff --git a/drivers/nxp/s32/ddr/s32g3/pie_cfg.c b/drivers/nxp/s32/ddr/s32g3/pie_cfg.c
index d0ec60202..797d51416 100644
--- a/drivers/nxp/s32/ddr/s32g3/pie_cfg.c
+++ b/drivers/nxp/s32/ddr/s32g3/pie_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf_16 pie_cfg[] = {
 	{0x40380c04, 0x00000010},
diff --git a/drivers/nxp/s32/ddr/s32r/ddrc_cfg.c b/drivers/nxp/s32/ddr/s32r/ddrc_cfg.c
index 18bafa48b..639c25245 100644
--- a/drivers/nxp/s32/ddr/s32r/ddrc_cfg.c
+++ b/drivers/nxp/s32/ddr/s32r/ddrc_cfg.c
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf ddrc_cfg[] = {
 	{0x4007c604, 0x00000000U},
diff --git a/drivers/nxp/s32/ddr/s32r/dmem_cfg.c b/drivers/nxp/s32/ddr/s32r/dmem_cfg.c
index a26a4c478..2898ea2c1 100644
--- a/drivers/nxp/s32/ddr/s32r/dmem_cfg.c
+++ b/drivers/nxp/s32/ddr/s32r/dmem_cfg.c
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 uint16_t dmem_1d_cfg[] = {
 	0x0000,
diff --git a/drivers/nxp/s32/ddr/s32r/dq_swap_cfg.c b/drivers/nxp/s32/ddr/s32r/dq_swap_cfg.c
index b23a12d15..c1a6dab52 100644
--- a/drivers/nxp/s32/ddr/s32r/dq_swap_cfg.c
+++ b/drivers/nxp/s32/ddr/s32r/dq_swap_cfg.c
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct dqconf dq_swap_cfg[] = {
 	{0x40394830, 0x02},
diff --git a/drivers/nxp/s32/ddr/s32r/phy_cfg.c b/drivers/nxp/s32/ddr/s32r/phy_cfg.c
index 3a7f4aff5..aa94df91a 100644
--- a/drivers/nxp/s32/ddr/s32r/phy_cfg.c
+++ b/drivers/nxp/s32/ddr/s32r/phy_cfg.c
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf_16 phy_cfg[] = {
 	{0x4038040c, 0x00000001},
diff --git a/drivers/nxp/s32/ddr/s32r/pie_cfg.c b/drivers/nxp/s32/ddr/s32r/pie_cfg.c
index ecf2beb1d..b9a136967 100644
--- a/drivers/nxp/s32/ddr/s32r/pie_cfg.c
+++ b/drivers/nxp/s32/ddr/s32r/pie_cfg.c
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 
 struct regconf_16 pie_cfg[] = {
 	{0x40380c04, 0x00000010},
diff --git a/plat/nxp/s32/s32_bl2_el3.c b/plat/nxp/s32/s32_bl2_el3.c
index 7dc843a27..5fbfdbdd4 100644
--- a/plat/nxp/s32/s32_bl2_el3.c
+++ b/plat/nxp/s32/s32_bl2_el3.c
@@ -11,7 +11,7 @@
 #include <common/fdt_fixup.h>
 #include <common/fdt_wrappers.h>
 #include <ddr/ddr_density.h>
-#include <ddr/ddr_utils.h>
+#include "ddr_utils.h"
 #include <lib/libc/errno.h>
 #include <lib/libfdt/libfdt.h>
 #include <lib/mmio.h>
diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index 3e4b2636f..803a203ca 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -44,6 +44,7 @@ PLAT_INCLUDES 	+= \
 			-Iinclude/common/tbbr \
 			-Iinclude/drivers \
 			-Iinclude/${S32_DRIVERS} \
+			-Iinclude/${S32_DRIVERS}/ddr \
 			-Iinclude/lib \
 			-Iinclude/lib/libc \
 			-Iinclude/lib/psci \
diff --git a/plat/nxp/s32/s32_ddr.mk b/plat/nxp/s32/s32_ddr.mk
index 1f7f97dc9..afe1224b3 100644
--- a/plat/nxp/s32/s32_ddr.mk
+++ b/plat/nxp/s32/s32_ddr.mk
@@ -43,3 +43,9 @@ import_ddr_drv: ${DDR_UTILS_FILE}
 	@${SED} -ie "s/\(#define RETENTION_ADDR\).*STNDBY_RAM_BASE/\1 BL31SSRAM_CSR_BASE/g" $<
 	@${ECHO} "  EDITED  $<"
 endif
+
+ifneq (${CUSTOM_DDR_DRV},)
+PLAT_INCLUDES += \
+	-I ${CUSTOM_DDR_DRV} \
+
+endif
\ No newline at end of file
diff --git a/plat/nxp/s32/s32g/bl31_sram/bl31_sram.mk b/plat/nxp/s32/s32g/bl31_sram/bl31_sram.mk
index 86089d178..44966e705 100644
--- a/plat/nxp/s32/s32g/bl31_sram/bl31_sram.mk
+++ b/plat/nxp/s32/s32g/bl31_sram/bl31_sram.mk
@@ -9,7 +9,7 @@ BL31SRAM_SOURCES = plat/common/aarch64/platform_up_stack.S \
 		   plat/nxp/s32/s32g/bl31_sram/bl31sram_main.c \
 		   plat/nxp/s32/s32g/s32g_clocks.c \
 		   plat/nxp/s32/s32g/s32g_mc_me.c \
-		   ${DDR_DRV}/ddr_lp.c \
+		   ${COMMON_DDR_DRV}/ddr_lp.c \
 
 BL31SRAM_ARRAY_NAME ?= bl31sram
 BL31SRAM_ARRAY_LEN  ?= bl31sram_len
diff --git a/plat/nxp/s32/s32g/bl31_ssram/bl31ssram_main.c b/plat/nxp/s32/s32g/bl31_ssram/bl31ssram_main.c
index 3535e0519..3740e5794 100644
--- a/plat/nxp/s32/s32g/bl31_ssram/bl31ssram_main.c
+++ b/plat/nxp/s32/s32g/bl31_ssram/bl31ssram_main.c
@@ -5,7 +5,7 @@
  */
 #include <common/debug.h>
 #include <plat/common/platform.h>
-#include <drivers/nxp/s32/ddr/ddr_lp.h>
+#include "ddr_lp.h"
 #include <platform_def.h>
 #include <s32g_clocks.h>
 #include <plat/nxp/s32g/bl31_ssram/ssram_mailbox.h>
diff --git a/plat/nxp/s32/s32g/s32g_bl2_el3.c b/plat/nxp/s32/s32g/s32g_bl2_el3.c
index bd407c2c3..abe0802b9 100644
--- a/plat/nxp/s32/s32g/s32g_bl2_el3.c
+++ b/plat/nxp/s32/s32g/s32g_bl2_el3.c
@@ -20,10 +20,10 @@
 #include <plat/nxp/s32g/bl31_ssram/ssram_mailbox.h>
 #include "s32_sramc.h"
 #if S32CC_EMU == 1
-#include <ddr/ddrss.h>
+#include "ddrss.h"
 #else
-#include <ddr/ddr_init.h>
-#include <drivers/nxp/s32/ddr/ddr_lp.h>
+#include "ddr_init.h"
+#include "ddr_lp.h"
 #include <dt-bindings/ddr-errata/s32-ddr-errata.h>
 #endif
 
diff --git a/plat/nxp/s32/s32r/s32r_bl2_el3.c b/plat/nxp/s32/s32r/s32r_bl2_el3.c
index 5e943f694..63a90af6f 100644
--- a/plat/nxp/s32/s32r/s32r_bl2_el3.c
+++ b/plat/nxp/s32/s32r/s32r_bl2_el3.c
@@ -5,7 +5,7 @@
  */
 
 #include <common/desc_image_load.h>
-#include <ddr/ddr_init.h>
+#include "ddr_init.h"
 #include <lib/libc/errno.h>
 #include <lib/mmio.h>
 #include "platform_def.h"
-- 
2.25.1

